clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClient.java

```diff
@@ -95,6 +95,16 @@ public RequestConfig.Builder newRequestConfig() {
     return RequestConfig.copy(defaultRequestConfig);
   }
 
+  public <HttpT, RespT> void get(
+      final String path,
+      final RequestConfig requestConfig,
+      final Class<HttpT> responseType,
+      final JsonResponseTransformer<HttpT, RespT> transformer,
+      final HttpCamundaFuture<RespT> result) {
+    sendRequest(Method.GET, path, null, requestConfig, responseType, transformer, result);
+  }
```

Comment:
💭 I am wondering: should we tackle the "extends" mechanism the other way around, i.e. `HttpZeebeFuture extends HttpCamundaFuture`?

More details:
- All implementation code sits in the new `Camunda...` classes and moves from the `io.camunda.zeebe.client` package to `io.camunda.client`.
- All `Zeebe...` classes would extend the Camunda classes so they still exist for easier migration. If Zeebe classes are not needed externally, we don't extend the respective Camunda classes.
- Under the hood, all code already works with Camunda classes.
- With 8.8, we can remove the `io.camunda.zeebe.client` package in one swoop, no further adjustments should be necessary.
- Methods like those in the `HttpClient` could be called with a `HttpCamundaFuture` or `HttpZeebeFuture`. The method signature changes to use the `HttpCamundaFuture`. A deprecated `HttpZeebeFuture` could be handed in since it fulfills the contract, i.e. it is a `HttpCamundaFuture`.

I might overlook something that makes the transition for people harder, of course 🙂

Reply:
I will try to follow this approach and let you know 👍

Reply:
It seems more complex then I thought

Reply:
I was fearing that would happen, yeah 🙈

Reply:
Keeping the deprecated methods and adding the ones for the new type work just as well, of course. However, since this is an internal class, I was wondering if we actually need to keep the deprecated methods at all or if we can simply adjust all calling methods and use the new `HttpCamundaFuture` there.

Reply:
Yes, it make more sense to just use the `old methods` with `HttpCamundaFuture `

Reply:
Sounds like the cleanest solution, if possible.